DEVELOPMENT.md

```diff
@@ -415,6 +416,67 @@ in the actual implementation and give great flexibility in the tests, while mock
 tend to result in more code re-use and slim down the tests. The mocking implementation of a
 utils interface can facilitate implementations of related functions to be based on shared data.
 
+### Test Parallelization
+
+Tests that can be executed in parallel should be marked as such. With the command `t.Parallel()` the test framework can
+be notified that this test can run in parallel, and it can start running the next test.
+([Example in Stackoverflow](https://stackoverflow.com/questions/44325232/are-tests-executed-in-parallel-in-go-or-one-by-one))
+Therefore, this command shall be called at the beginning of a test method and `t.Run()` sub tests.
+See also the [documentation](https://golang.org/pkg/testing/#T.Parallel) for `t.Parallel()` and `t.Run()`.
+
+```go
+func TestMethod(t *testing.T) {
+    t.Parallel() // indicates that this method can parallel to other methods
+
+    t.Run("sub test 1", func(t *testing.T){
+        t.Parallel() // indicates that this sub test can run parallel to other sub tests
+        // execute test
+    })
+
+    t.Run("sub test 2", func(t *testing.T){
+        t.Parallel() // indicates that this sub test can run parallel to other sub tests
+        // execute test
+    })
+}
+```
+
+Go will first execute the non-parallelized tests in sequence and afterwards execute the all parallel tests in parallel
+limited by the default number of parallel executions.
+
+For table tests some additional lines are necessary to prevent overshadowing of the iteration variable.
```

Comment:
```suggestion
It is important that tests executed in parallel use the variable values actually meant to be visible to them.
Especially in table tests, it can happen easily that a variable injected into the `t.Run()`-closure via the outer scope is changed before or while the closure executes.
To prevent this, it is possible to create shadowing instances of variables in the body of the test loop.
```

Reply:
Sounds better :-)